app/src/main/java/org/p2p/wallet/user/repository/prices/impl/TokenPricesCoinGeckoRepository.kt

```diff
@@ -36,4 +37,30 @@ class TokenPricesCoinGeckoRepository(
             )
                 .map { TokenPrice(tokenId = it.id, price = it.currentPrice) }
         }
+
+    override suspend fun getTokenPricesByAddressesMap(
+        tokenAddresses: List<TokenAddress>,
+        targetCurrency: String
+    ): Map<TokenAddress, TokenPrice> {
+        return loadEthereumPrices(tokenAddresses, targetCurrency).associateBy { TokenAddress(it.tokenId) }
+    }
+
+    override suspend fun getTokenPriceByAddress(
+        tokenAddress: TokenAddress,
+        targetCurrency: String
+    ): TokenPrice = loadEthereumPrices(listOf(tokenAddress), targetCurrency).first()
+
+    private suspend fun loadEthereumPrices(
+        tokenAddresses: List<TokenAddress>,
+        targetCurrencySymbol: String
+    ): List<TokenPrice> =
+        withContext(dispatchers.io) {
+            val tokenAddressesForRequest = tokenAddresses.joinToString(",") { it.address }
+            coinGeckoApi.getEthereumTokenPrices(
+                tokenAddresses = tokenAddressesForRequest,
+                targetCurrency = targetCurrencySymbol.lowercase()
+            ).map { (tokenAddress, tokenPrice) ->
+                TokenPrice(tokenId = tokenAddress, price = tokenPrice.currentPrice)
```

Comment:
we can be confused later when using TokenPrice class
everywhere for SOL tokenId for us is coingeckoId or some id, not address
maybe create own wrapper class for Eth prices?

Reply:
I Remove this methods from coingecko, so it's out of date